### PR TITLE
Adds an index file to each directory, and removes gz suffix

### PR DESCRIPTION
**Pull request by jneem**
The index file contains an line of the form `{name:"394250", min_timestamp: 8509, max_timestmap: 2039481049}` for each element in the directory.

### CODE REVIEW

File: firehose.go
Comment (chongr) [inline firehose.go]: https://github.com/cloudchacho/hedwig-firehose/blob/main/gcp/gcp.go#L105

Think you can use the precondition to make a reader if it doesn't exist.  Potentially could be another error on createReader and might lose the read data?

File: firehose.go
Comment (chongr) [inline firehose.go:348]: hmmm we may have to just try this wasn't able to get write down to 5 minutes so hopefully this file doesn't get too big

File: firehose.go
Comment (jneem) [reply firehose.go]: Thanks for the pointer! I didn't manage to make it work, though: adding `If(storage.Conditions{DoesNotExist: false})` didn't have any effect. (And then I [found](https://pkg.go.dev/cloud.google.com/go/storage#Conditions) that it isn't supposed to do anything.)

I did improve the checking so that other errors won't cause issues.

File: firehose.go
Comment (jneem) [reply firehose.go:348]: Brandon pointed out that there's a more complicated scheme that we can use if the index gets big: after some threshold, we just start a new empty index file. The new index file contains a line pointing to the GCS generation of the previous index file, and so then on reading we can "follow the links" and concatenate the files together.

The nice thing about the more complicated scheme is that it's forwards-compatible with data written under the simple scheme. So we can see if it's a problem first and then do the more complicated thing if it is.

Comment (chongr) [reply firehose.go:348]: ah I see makes sense yeah we also could just number them

**Review by chongr — DISMISSED** [review]

**Comment by chongr** [comment]
might need a test when ndjson file is already there too

**Comment by jneem** [comment]
Oh good point, I'll add that. Maybe it will make the codecov check happy, too :crossed_fingers:

**Review by chongr — APPROVED** [review]

**Comment by jneem** [comment]
Is there something else I need to do here? (if you're waiting for me to hit the merge button, I don't have permissions for that :smiley: )

**Comment by chongr** [comment]
my bad - didn't realize

**Pull request merged**
